### ISSUE TITLE
❓How train model with standard for loop?

### ISSUE THREAD

**Issue by GiovanniPasq** [facebookresearch/detectron2#1155]
Hi, I trained my model following the colab tutorial without any problem, however when i try to train it with the standard for loop the loss of my model become nan. Can anyone help me?
I used the following code (it came from the plain_train_net.py script):

```
register_coco_instances("dataset")
cfg = get_cfg()
cfg.merge_from_file(model_zoo.get_config_file("COCO-Detection/retinanet_R_101_FPN_3x.yaml"))
cfg.DATASETS.TRAIN = ("dataset",)
cfg.DATALOADER.NUM_WORKERS = 0
cfg.SOLVER.IMS_PER_BATCH = 1
cfg.MODEL.WEIGHTS = model_zoo.get_checkpoint_url("COCO-Detection/retinanet_R_101_FPN_3x.yaml")
cfg.MODEL.RETINANET.NUM_CLASSES = 16
cfg.SOLVER.BASE_LR = 0.0001
cfg.SOLVER.MAX_ITER = 10000
cfg.MODEL.DEVICE = "cuda:0"
model = build_model(cfg)

model.train()
optimizer = build_optimizer(cfg, model)
scheduler = build_lr_scheduler(cfg, optimizer)

data_loaderTrainLabeled = build_detection_train_loader(cfg)
with EventStorage() as storage:
  for dataLabeled, iteration in zip(data_loaderTrainLabeled, range(0,cfg.SOLVER.MAX_ITER)):
              iteration = iteration + 1 
              loss_dict = model(dataLabeled)
              losses = sum(loss_dict.values())
              optimizer.zero_grad()
              losses.backward()
              optimizer.step()
              scheduler.step()


Output when I print loss_dict:
{'loss_cls': tensor(1.4841e+08, device='cuda:0', grad_fn=<DivBackward0>), 'loss_box_reg': tensor(14263.7764, device='cuda:0', grad_fn=<DivBackward0>)}
{'loss_cls': tensor(nan, device='cuda:0', grad_fn=<DivBackward0>), 'loss_box_reg': tensor(0.2823, device='cuda:0', grad_fn=<DivBackward0>)}

**Comment by ppwwyyxx** [comment]
https://detectron2.readthedocs.io/tutorials/training.html has mentioned training examples using standard for loop. 

> You are free to create your own optimizer, and write the training logic: it’s usually easy with PyTorch, and allow researchers to see the entire training logic more clearly and have full control. One such example is provided in tools/plain_train_net.py.
